src/js/polyfills/console.js

```diff
@@ -1,258 +1,316 @@
 /* global tjs */
+/** console implemenation according to https://console.spec.whatwg.org/ and otherwise inspired by nodejs */
 
-import { format } from 'util';
+import { format as utilFormat, inspect as utilInspect } from 'util';
 
-const encoder = new TextEncoder();
-
-function print() {
-    const text = format.apply(null, arguments) + '\n';
-
-    tjs.stdout.write(encoder.encode(text));
-}
+function createConsole({ logger, clear, printer, formatter, inspect }) {
+    if (!printer) {
+        throw new Error('Printer is required');
+    }
 
+    const _printer = (logLevel, ...args) => {
+        printer(logLevel, groupCount, args);
+    };
 
-// Copyright Joyent, Inc. and other Node contributors. MIT license.
-// Forked from Node's lib/internal/cli_table.js
+    if (!logger) {
+        logger = function Logger(logLevel, args) {
+            if (args.length === 1) {
+                _printer(logLevel, args[0]);
```

Comment:
Why is the single argument not formatted here?

Reply:
See spec: https://console.spec.whatwg.org/#logger

Reply:
👍